pkg/api/api.go

```diff
@@ -268,6 +268,7 @@ func (hs *HTTPServer) registerRoutes() {
 		// Preferences
 		apiRoute.Group("/preferences", func(prefRoute routing.RouteRegister) {
 			prefRoute.Post("/set-home-dash", routing.Wrap(hs.SetHomeDashboard))
+			prefRoute.Put("/set-json-data", routing.Wrap(hs.SetPreferencesJsonData))
```

Comment:
This one doesn't feel optimal introducing a specific new endpoint for only updating navbar preferences when we already have the following:
- https://grafana.com/docs/grafana/latest/http_api/preferences/
- https://grafana.com/docs/grafana/latest/http_api/team/#get-team-preferences

No blocker for merging, but before release it would be nice to try make this a bit more consistent by
- Updating [getPreferencesFor](https://github.com/grafana/grafana/blob/10b3847d57dafd42eeb18f89a1d531fc569225cd/pkg/api/preferences.go#L40) to return JSONData preferences and/or each property in JSONData as it's own root property, similar as theme, homedashboardId etc.
- Reuse [updatePreferencesFor](https://github.com/grafana/grafana/blob/10b3847d57dafd42eeb18f89a1d531fc569225cd/pkg/api/preferences.go#L66) to allow saving JSONData preferences - if JSONData preferences not provided, don't update it to keep backward compatibility.
- Introducing a `patchPreferencesFor` method to support HTTP `PATCH` verb to only update provided preferences. With these changes I think we can deprecate `/set-home-dash` since it doesn't seem to be used by Grafana UI and then remove it completely in a major version.